manager/manager.go

```diff
@@ -746,6 +746,9 @@ func (m *manager) registerCollectors(collectorConfigs map[string]string, cont *c
 
 // Create a container.
 func (m *manager) createContainer(containerName string) error {
+	m.containersLock.Lock()
+	defer m.containersLock.Unlock()
+
 	handler, accept, err := container.NewContainerHandler(containerName, m.inHostNamespace)
```

Comment:
If we were to check if the container exists before invoking this Factory method, wouldn't that be better?